samples/snippets/src/main/java/com/example/bigquerystorage/WriteToDefaultStream.java

```diff
@@ -135,22 +135,29 @@ private static class DataWriter {
     // Track the number of in-flight requests to wait for all responses before shutting down.
     private final Phaser inflightRequestCount = new Phaser(1);
     private final Object lock = new Object();
+
     private JsonStreamWriter streamWriter;
 
     @GuardedBy("lock")
     private RuntimeException error = null;
 
     public void initialize(TableName parentTable)
         throws DescriptorValidationException, IOException, InterruptedException {
+      initialize(parentTable.toString());
+    }
+
+    private void initialize(String streamOrTableName)
+        throws DescriptorValidationException, IOException, InterruptedException {
       // Use the JSON stream writer to send records in JSON format. Specify the table name to write
       // to the default stream.
       // For more information about JsonStreamWriter, see:
       // https://googleapis.dev/java/google-cloud-bigquerystorage/latest/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.html
       streamWriter =
-          JsonStreamWriter.newBuilder(parentTable.toString(), BigQueryWriteClient.create()).build();
+          JsonStreamWriter.newBuilder(streamOrTableName, BigQueryWriteClient.create())
+              .build();
     }
 
-    public void append(AppendContext appendContext)
+    public synchronized void append(AppendContext appendContext)
```

Comment:
why this need to be changed to synchronized?

Reply:
Because we could be replacing the writer in the function. The other way is to make the writer be protected by a lock, but that would require writer.append be protected by a lock too.

Reply:
I removed the synchronize since I found the append is already in a lock below.